### PR TITLE
introduce schema v2beta24

### PR DESCRIPTION
**Pull request by ahmetb**
**Related**: #6489

### CODE REVIEW

File: pkg/skaffold/schema/v2beta23/upgrade.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+	pkgutil.CloneThroughJSON(c, &newConfig)
+	newConfig.APIVersion = next.Version
+
+	err := util.UpgradePipelines(c, &newConfig, upgradeOnePipeline)
+	return &newConfig, err
+}
+
+func upgradeOnePipeline(oldPipeline, newPipeline interface{}) error {
+	// oldBuild := &oldPipeline.(*Pipeline).Build
+	// newBuild := &newPipeline.(*next.Pipeline).Build
```
Comment (MarlonGamez) [inline pkg/skaffold/schema/v2beta23/upgrade.go:38]: Think we can just remove these commented lines

**Review by MarlonGamez — COMMENTED** [review]
looks good, have a small nit

**Review by MarlonGamez — APPROVED** [review]
Looks good to me ✅

**Comment by ahmetb** [comment]
I've got Kokoko reproducibly failing on an interactive test (`TestDevGracefulCancel/getting-started` tracked in #6424).

I don't think I've done anything notable other than clearing the `upgrade.go` contents, and fixing the `upgrade_test.go` to have identical `expected`/`want` YAMLs. I used the instructions in DEVELOPMENT.md.

https://source.cloud.google.com/results/invocations/0ae17eca-0fda-4f28-8947-d30f12d0efa7/targets/skaffold%2Fpresubmit/log

**Comment by MarlonGamez** [comment]
@ahmetb I will merge this for now since kokoro is so inconsistent

**Pull request merged**